### PR TITLE
Daemon/Service heartbeats, and Notification services

### PR DESCRIPTION
**Pull request by Dinesh2521**
@vsbogd  @raamb  @anandrgit  please review the changes for adding heartbeat functionality to Daemon (wraps service heartbeat as well) along with Notification/Alerts functionality.
more details are available in metrics/README.md

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -153,6 +166,20 @@ func Validate() error {
 		return errors.New("SSL requires both key and certificate when enabled")
 	}
 
+	// validate heartbeat, notification, and monitoring service endpoints
```
Comment (raamb) [inline config/config.go]: For notification and service hearbeat endpoints lets do the following
1) If no endpoint is given then lets ignore this functionality
2) If an endpoint is given and its invalid then lets stop the daemon. 
Essentially we are saying that if a URL is specified then it must be a valid one, no URL is treated as disabling the functionality

For monitoring endpoint the validity check should be based on if monitoring is enabled or not.

File: resources/blockchain/package-lock.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
       "resolved": "https://registry.npmjs.org/abstract-leveldown/-/abstract-leveldown-2.6.3.tgz",
       "integrity": "sha512-2++wDf/DYqkPR3o5tbfdhF96EfMApo1GpPfzOsR/ZYXdkSmELlvOOEAl9iKkRsktMPHdGjO4rtkBpf2I7TiTeA==",
       "requires": {
-        "xtend": "4.0.1"
+        "xtend": "~4.0.0"
```
Comment (raamb) [inline resources/blockchain/package-lock.json]: Do we need to go to an older version?

File: resources/blockchain/package-lock.json [hunk 3]
```diff
@@ -54,15 +71,15 @@
...
+        "lodash": "^4.17.10"
       }
     },
     "async-eventemitter": {
       "version": "0.2.4",
       "resolved": "https://registry.npmjs.org/async-eventemitter/-/async-eventemitter-0.2.4.tgz",
       "integrity": "sha512-pd20BwL7Yt1zwDFy+8MX8F1+WCT8aQeKj0kQnTrH9WaeRETlRamVhD0JtRPmrV4GfOJ2F9CvdQkZeZhnh2TuHw==",
       "requires": {
-        "async": "2.6.1"
+        "async": "^2.4.0"
```
Comment (raamb) [inline resources/blockchain/package-lock.json]: Same here and in other places in this file.

File: resources/blockchain/package-lock.json (hunk 2 above)
Comment (Dinesh2521) [reply resources/blockchain/package-lock.json]: please ignore this file. It supposed to be not checked-in

File: config/config.go (hunk 1 above)
Comment (Dinesh2521) [reply config/config.go]: fixed the issue. please review

File: resources/blockchain/package-lock.json (hunk 3 above)
Comment (Dinesh2521) [reply resources/blockchain/package-lock.json]: this file not supposed to be checked-in. please ignore it.

File: config/config.go [hunk 4]
```diff
@@ -42,6 +44,12 @@ const (
 	PaymentChannelStorageTypeKey   = "payment_channel_storage_type"
 	PaymentChannelStorageClientKey = "payment_channel_storage_client"
 	PaymentChannelStorageServerKey = "payment_channel_storage_server"
+	//configs for Daemon Monitoring and Notification
+	AlertsEMail                 = "alerts_email"
+	HeartbeatServiceEndpoint    = "heartbeat_svc_end_point"
+	NotificationServiceEndpoint = "notification_svc_end_point"
+	ServiceHeartbeatType        = "service_heartbeat_type"
```
Comment (vsbogd) [inline config/config.go]: Could you please add documentation for new fields into daemon README.md (https://github.com/singnet/snet-daemon#configuration). Also two new fields from previous PR.

Comment (vsbogd) [reply config/config.go]: Ah, I see that metrics documentation is added to metrics package. Then could you please add the link to it into main configuration secition https://github.com/singnet/snet-daemon#configuration

Comment (Dinesh2521) [reply config/config.go]: Updated and checked-in

File: resources/blockchain/package-lock.json (hunk 3 above)
Comment (raamb) [reply resources/blockchain/package-lock.json]: @Dinesh2521 please undo this change and update the PR, will make it easier to merge.

File: resources/blockchain/package-lock.json (hunk 2 above)
Comment (raamb) [reply resources/blockchain/package-lock.json]: @Dinesh2521 please undo this change and update the PR, will make it easier to merge.

Comment (Dinesh2521) [reply resources/blockchain/package-lock.json]: reverted the file to original state and checked in. Couldn't remove the file completely from PR.

File: resources/blockchain/package-lock.json (hunk 3 above)
Comment (Dinesh2521) [reply resources/blockchain/package-lock.json]: reverted the file to original state and checked in. Couldn't remove the file completely from PR.

**Pull request merged**